java/Indexer/src/main/java/MapIndexer.java

```diff
@@ -0,0 +1,47 @@
+import java.io.File;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+
+public class MapIndexer {
+
+    private Map<String, HashSet<Integer>> wordIndexes;
+    private DataMart dataMart;
+
+    public MapIndexer() {
+        wordIndexes = new HashMap<>();
```

Comment:
The whole `HashMap` here is unnecessary Imo. It creates another Data Mart basically. You can just use the `FileDataMart`. It automatically checks for duplicates. But perhaps I should mention it in the documentation.

Reply:
you are right. ill try to change it